SDWebImage/Core/SDImageCache.m

```diff
@@ -607,6 +620,7 @@ + (SDWebImageOptions)imageOptionsFromCacheOptions:(SDImageCacheOptions)cacheOpti
     if (cacheOptions & SDImageCacheDecodeFirstFrameOnly) options |= SDWebImageDecodeFirstFrameOnly;
     if (cacheOptions & SDImageCachePreloadAllFrames) options |= SDWebImagePreloadAllFrames;
     if (cacheOptions & SDImageCacheAvoidDecodeImage) options |= SDWebImageAvoidDecodeImage;
+    if (cacheOptions & SDImageCacheMatchAnimatedImageClass) options |= SDWebImageMatchAnimatedImageClass;
```

Comment:
Should we deal with like below:
```ruby
if (cacheOptions & SDImageCacheDecodeFirstFrameOnly) {
     options |= SDWebImageDecodeFirstFrameOnly;
     options &= ~SDImageCacheMatchAnimatedImageClass;
}

Reply:
No, we don't. My comment in header is about `what's the behavior of these two options`, but not means `we must remove one if they conflict`.

In every implementation file, when we see `firstFrameOnly`, we use `NSImage/UIImage`. So this is not related to any custom animated image class.